test/reactors/V3DutchOrderReactor.t.sol

```diff
@@ -0,0 +1,1005 @@
+            exclusivityOverrideBps: 0,
+            // override is more input tokens than expected
+            inputAmount: 1 ether,
+            outputAmounts: ArrayBuilder.fill(1, 0.9 ether)
+        });
+
+        V3DutchOrder memory order = V3DutchOrder({
+            info: OrderInfoBuilder.init(address(reactor)).withSwapper(swapper),
+            cosigner: vm.addr(cosignerPrivateKey),
+            baseInput: V3DutchInput(tokenIn, 1 ether, CurveBuilder.emptyCurve(), 1 ether),
+            baseOutputs: OutputsBuilder.singleV3Dutch(
+                address(tokenOut), 1 ether, CurveBuilder.singlePointCurve(1, 0.2 ether), swapper
+            ),
+            cosignerData: cosignerData,
+            cosignature: bytes("")
+        });
+        order.cosignature = cosignOrder(order.hash(), cosignerData);
+        SignedOrder memory signedOrder =
+            SignedOrder(abi.encode(order), signOrder(swapperPrivateKey, address(permit2), order));
+        vm.expectRevert(V3DutchOrderReactor.InvalidCosignerOutput.selector);
+        fillContract.execute(signedOrder);
+    }
+
+    function testV3OutputOverrideWrongLength() public {
+        CosignerData memory cosignerData = CosignerData({
+            decayStartBlock: block.number,
+            exclusiveFiller: address(0),
+            exclusivityOverrideBps: 0,
+            // override is more input tokens than expected
+            inputAmount: 1 ether,
+            outputAmounts: ArrayBuilder.fill(2, 1.1 ether)
+        });
+
+        V3DutchOrder memory order = V3DutchOrder({
+            info: OrderInfoBuilder.init(address(reactor)).withSwapper(swapper),
+            cosigner: vm.addr(cosignerPrivateKey),
+            baseInput: V3DutchInput(tokenIn, 1 ether, CurveBuilder.emptyCurve(), 1 ether),
+            baseOutputs: OutputsBuilder.singleV3Dutch(
+                address(tokenOut), 1 ether, CurveBuilder.singlePointCurve(1, 0.2 ether), swapper
+            ),
+            cosignerData: cosignerData,
+            cosignature: bytes("")
+        });
+        order.cosignature = cosignOrder(order.hash(), cosignerData);
+        SignedOrder memory signedOrder =
+            SignedOrder(abi.encode(order), signOrder(swapperPrivateKey, address(permit2), order));
+        vm.expectRevert(V3DutchOrderReactor.InvalidCosignerOutput.selector);
+        fillContract.execute(signedOrder);
+    }
+
+    function testV3OverrideInput() public {
+        uint256 outputAmount = 1 ether;
+        uint256 overriddenInputAmount = 0.7 ether;
+        tokenIn.mint(swapper, overriddenInputAmount);
+        tokenOut.mint(address(fillContract), outputAmount);
+        tokenIn.forceApprove(swapper, address(permit2), type(uint256).max);
+        CosignerData memory cosignerData = CosignerData({
+            decayStartBlock: block.number,
+            exclusiveFiller: address(0),
+            exclusivityOverrideBps: 0,
+            inputAmount: overriddenInputAmount,
+            outputAmounts: ArrayBuilder.fill(1, 0)
```

Comment:
nit: not sure if this is the same from V2DutchOrder tests but the gas on this will be off due to the 0 transfer of the output

Reply:
[Yeah it's similar to V2](https://github.com/Uniswap/UniswapX/blob/b3a807b589a2b8fe2220fdab44ea66eaf2ce2e7e/test/reactors/V2DutchOrderReactor.t.sol#L364). I've been using the gas measurements more just to get a sense of the relative changes.